hibernate-core/src/main/java/org/hibernate/boot/model/source/internal/hbm/ModelBinder.java

```diff
@@ -3603,6 +3616,11 @@ else if ( getPluralAttributeSource().getElementSource() instanceof PluralAttribu
 
 				);
 				getCollectionBinding().setElement( elementBinding );
+				// Collection#setWhere is used to set the "where" clause that applies to the collection table
+				// (which is the join table for a many-to-any association).
+				// This "where" clause comes from the collection mapping; e.g., <set name="..." ... where="..." .../>
+				getCollectionBinding().setWhere( getPluralAttributeSource().getWhere() );
+				getCollectionBinding().setWhere( getPluralAttributeSource().getWhere() );
```

Comment:
This line is duplicated.

Reply:
Ah yes, must have done that when I copied the comment at the last minute. I've removed it.